### PR TITLE
fix dialog default enter behaviour + sort elements in alphabetical order in directory-tree & directory-content

### PR DESCRIPTION
**Pull request by ChamseddineBhd**
Signed-off-by: chamseddine <chamseddine.benhamed@ensi-uma.tn>

### CODE REVIEW

File: src/components/dialogs/delete-dialog.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ const DeleteDialog = ({ open, onClose, onClick, title, message, error }) => {
 
     const handleKeyPressed = (event) => {
         if (open && event.key === 'Enter') {
-            handleClick();
+            handleClose();
```
Comment (sBouzols) [inline src/components/dialogs/delete-dialog.js:39]: Not sure here... I continue to think it should be handleClick even if its a delete dialog

File: src/components/tree-views-container.js [hunk 2]
```diff
@@ -161,9 +161,11 @@ const TreeViewsContainer = () => {
         (children) => {
             dispatch(
                 setCurrentChildren(
-                    children.filter(
-                        (child) => child.type !== elementType.DIRECTORY
-                    )
+                    children
```
Comment (sBouzols) [inline src/components/tree-views-container.js:168]: Not sure if it's the right place to sort... I think it's better to do it during rendering like in renderTree for TreeItems...
Maybe move this sorting code somewhere in directory-content render method ?

File: src/components/directory-tree-view.js [hunk 3]
```diff
@@ -250,9 +248,13 @@ const DirectoryTreeView = ({ treeViewUID, mapData }) => {
                 }}
             >
                 {Array.isArray(node.children)
-                    ? node.children.map((child) =>
-                          renderTree(mapDataRef.current[child.elementUuid])
-                      )
+                    ? node.children
```
Comment (sBouzols) [inline src/components/directory-tree-view.js]: The sorting will not work for rootDirectories then (the first TreeItem in this recursive function)... we must add this sorting code in directory-container when we loop over rootdirectories to render TreeViews.

File: src/components/directory-tree-view.js [hunk 4]
```diff
@@ -153,7 +153,6 @@ const DirectoryTreeView = ({ treeViewUID, mapData }) => {
 
     const handleOpenMenu = (event) => {
         setAnchorEl(event.currentTarget);
-        event.stopPropagation();
```
Comment (sBouzols) [inline src/components/directory-tree-view.js:156]: I don't see any differences then I imagine it's a good pratice to remove code which we don't understand behavior but maybe you could explain here why `event.stopPropagation()` and `e.preventDefault()` are useless

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/directory-tree-view.js (hunk 3 above)
Comment (ChamseddineBhd) [reply src/components/directory-tree-view.js]: it's already added

File: src/components/directory-tree-view.js (hunk 4 above)
Comment (ChamseddineBhd) [reply src/components/directory-tree-view.js:156]: because we have added  event.preventDefault() in app.js in a previous commit

**Review by sBouzols — APPROVED** [review]

**Pull request merged**